packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -770,6 +759,20 @@ const ComboBox = forwardRef(
           }
         }
       },
+      onStateChange: ({ type, selectedItem: newSelectedItem }) => {
+        if (
+          type === '__item_click__' &&
+          !isEqual(selectedItemProp, newSelectedItem)
+        ) {
+          onChange({ selectedItem: newSelectedItem });
+        }
+        if (
+          type === '__function_select_item__' ||
```

Comment:
Should use enum

Reply:
This pull request is already merged, and a fix for this has also been merged, see https://github.com/carbon-design-system/carbon/issues/18145